### DISCUSSION TITLE
Jolokia Integration project

### DISCUSSION THREAD

**Discussion by grgrzybek** [rhuss/jolokia#757] [Ideas]
After implementing #564, where
* _data updater_ became a discoverable service
* _cache key provider_ was introduced for list caching purposes (optimization of huge responses)

I found several scenarios, where integration with other projects would be beneficial both for Jolokia and for these projects.

For example, when Artemis broker runs with hundreds or thousands of queues, we have a lot of MBeans with the same MBeanInfo - `list()` operation becomes very slow, because the JSON data for 10000 queues is over 200MB long. When this `list()` uses cached version if MBeanInfo for Artemis queue, the size drops below 1MB!

However I'd like to avoid adding _data updaters_ and _cache key providers_ in Jolokia project itself (the _core_ Jolokia project) and instead I suggest creating `jolokia/jolokia-integration` project, with different release cycle than `jolokia/jolokia`. It could contain Artemis (or Camel, or Tomcat or Jetty, ...) specific code.

**Comment by tadayosi** [comment]
Sounds great to me. An only concern is when we go the path, we'd need to maintain compatibility matrix between `jolokia/jolokia` and `jolokia/jolokia-integration`. But it shouldn't be a big deal.

**Reply by grgrzybek** [reply]
> Sounds great to me. An only concern is when we go the path, we'd need to maintain compatibility matrix between `jolokia/jolokia` and `jolokia/jolokia-integration`. But it shouldn't be a big deal.

Correct - a little effort, but it should be kind of provider-consumer relationship via API. And API changes should require at least minor (not micro) version changes.
